### PR TITLE
refactor(gcp): refactor GCP provider

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Refactor code from GCP provider to ensure that it can be used as a library from the SDK.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/gcp/gcp_provider.py [hunk 1]
```diff
@@ -38,13 +38,18 @@ class GcpProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments):
+    def __init__(
+        self,
+        input_project_ids,
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py:43]:
```suggestion
        project_ids,
```

Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py]:
```suggestion
        project_ids,
```

File: prowler/providers/gcp/gcp_provider.py [hunk 2]
```diff
@@ -38,13 +38,18 @@ class GcpProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments):
+    def __init__(
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py:41]: Please add default values and document the class.

File: prowler/providers/gcp/gcp_provider.py [hunk 3]
```diff
@@ -56,7 +61,7 @@ def __init__(self, arguments):
         accessible_projects = self.get_projects()
         if not accessible_projects:
             logger.critical("No Project IDs can be accessed via Google Credentials.")
-            sys.exit(1)
+            raise SystemExit
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py:77]: Add a TODO here to replace this with our custom exception.

File: prowler/providers/gcp/gcp_provider.py [hunk 4]
```diff
@@ -88,7 +93,7 @@ def __init__(self, arguments):
             logger.critical(
                 "No Input Project IDs can be accessed via Google Credentials."
             )
-            sys.exit(1)
+            raise SystemExit
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py:110]: Add a TODO here to replace this with our custom exception.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
There is something strange in this PR with the commits, please fix that just to have the code belonging to it.

File: prowler/providers/gcp/gcp_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/gcp/gcp_provider.py:41]: Done

File: prowler/providers/gcp/gcp_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/gcp/gcp_provider.py:77]: Done

File: prowler/providers/gcp/gcp_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/gcp/gcp_provider.py]: Done

File: prowler/providers/gcp/gcp_provider.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/gcp/gcp_provider.py:110]: Done

**Review by jfagoagas — APPROVED** [review]
👏

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**